src/masternodes/mn_checks.cpp

```diff
@@ -820,6 +820,10 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor {
         node.creationHeight      = height;
         node.operatorType        = obj.operatorType;
         node.operatorAuthAddress = obj.operatorAuthAddress;
+        if (height >= static_cast<uint32_t>(consensus.NextNetworkUpgradeHeight)) {
+            node.rewardAddressType   = obj.rewardType;
```

Comment:
In CMasternodesView::CreateMasternode add checks that the type is either NoDestType, PKHashType, ScriptHashType or WitV0KeyHashType, so 0, 1, 2 or 4.